test/optimizer/rule_test.cpp

```diff
@@ -54,5 +69,176 @@ TEST_F(RuleTests, SimpleRuleApplyTest) {
   EXPECT_EQ(outputs.size(), 1);
 }
 
+// Test whether update stament will use index scan plan
+TEST_F(RuleTests, UpdateDelWithIndexScanTest) {
```

Comment:
This test suite is for rule.cpp|h, please add a test suite purely for simple_optimizer and move this test there.